napari/layers/surface/surface.py

```diff
@@ -421,37 +498,69 @@ def _get_state(self):
+                'texture': self.texture,
+                'texcoords': self.texcoords,
+                'vertex_colors': self.vertex_colors,
             }
         )
         return state
 
+    def _slice_associated_data(
+        self,
+        data: np.ndarray,
+        vertex_ndim: int,
+        dims: int = 1,
+    ) -> Union[List[Any], np.ndarray]:
+        """Return associated layer data (e.g. vertex values, colors) within
+        the current slice.
+        """
+        if data is None:
+            return []
+
+        data_ndim = data.ndim - 1
+        if data_ndim >= dims:
+            # Get indices for axes corresponding to data dimensions
+            data_indices = self._slice_indices[:-vertex_ndim]
+            data = data[data_indices]
+            if data.ndim > dims:
+                warnings.warn(
+                    trans._(
+                        "Assigning multiple data per vertex after slicing "
+                        "is not allowed. All dimensions corresponding to "
+                        "vertex data must be non-displayed dimensions. Data "
+                        "may not be visible.",
+                        deferred=True,
+                    ),
+                    category=UserWarning,
+                    stacklevel=2,
+                )
+                return []
+        return data
+
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
         N, vertex_ndim = self.vertices.shape
         values_ndim = self.vertex_values.ndim - 1
 
-        # Take vertex_values dimensionality into account if more than one value
-        # is provided per vertex.
-        if values_ndim > 0:
-            # Get indices for axes corresponding to values dimensions
-            values_indices = self._slice_indices[:-vertex_ndim]
-            values = self.vertex_values[values_indices]
-            if values.ndim > 1:
-                warnings.warn(
-                    trans._(
-                        "Assigning multiple values per vertex after slicing is not allowed. All dimensions corresponding to vertex_values must be non-displayed dimensions. Data will not be visible.",
-                        deferred=True,
-                    )
-                )
-                self._data_view = np.zeros((0, self._slice_input.ndisplay))
-                self._view_faces = np.zeros((0, 3))
-                self._view_vertex_values = []
-                return
+        self._view_vertex_values = self._slice_associated_data(
+            self.vertex_values,
+            vertex_ndim,
+        )
 
-            self._view_vertex_values = values
-            # Determine which axes of the vertices data are being displayed
-            # and not displayed, ignoring the additional dimensions
-            # corresponding to the vertex_values.
+        # TODO: ensure vertex_colors matches dims with vertex_values
+        # OR is None OR is only one set
```

Comment:
Was this TODO addressed?

Reply:
Based on @jni's comments elsewhere in this PR, I had decided to "solve" this with better documentation and rely on users to set the values appropriately. This makes sense to me as it's not clear which order they may set vertex_values and vertex_colors. I removed this TODO and updated the docstring for the setter if you want to take a look and let me know what you think.

I guess one case we could handle is if vertex_values is None and vertex_colors is set with extra dimensions, it might be nice to have the sliders added based on the shape of verted_colors?